subsys/fs/nvs/nvs.c

```diff
@@ -386,20 +423,14 @@ static int nvs_prev_ate(struct nvs_fs *fs, uint32_t *addr, struct nvs_ate *ate)
 		return 0;
 	}
 
-	if (!nvs_ate_crc8_check(&close_ate)) {
-		/* update the address so it points to the last added ate.
-		 * do a check on close_ate.offset so that it does not point
-		 * outside a sector and is aligned to ate size.
-		 */
-		if (close_ate.offset < (fs->sector_size - ate_size) &&
-		    !(close_ate.offset % ate_size)) {
-			(*addr) &= ADDR_SECT_MASK;
-			(*addr) += close_ate.offset;
-			return 0;
-		}
+	/* Update the address if the close ate is valid.
+	 */
+	if (nvs_close_ate_valid(fs, &close_ate)) {
```

Comment:
lack of check of case `close_ate.offset > (fs->sector_size - ate_size)`
need add service of `close_ate.offset == (fs->sector_size - ate_size)`

Reply:
I don't think we need to service this, IMO the simplest idea is to consider any ate as valid if `.offset< fs->sector_size - ate_size` . In that case the service of the special case is automatically done.

Reply:
Almost, yes `.offset == (fs->sector_size - ate_size)` need special treatment